### PR TITLE
[ready] kokkos impl: `copy`, `idx_abs_max`

### PR DESCRIPTION
**Pull request by fnrizzi**

### CODE REVIEW

File: tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_copy_kk.hpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	    std::experimental::extents<ext_y ...>,
+	    Layout_y,
+	    std::experimental::default_accessor<ElementType_y>
+	  > y)
+requires ((sizeof...(ext_x) == sizeof...(ext_y)) && (x.rank() <=2))
+{
+  Impl::signal_kokkos_impl_called("copy");
+
+  auto x_view = Impl::mdspan_to_view(x);
+  auto y_view = Impl::mdspan_to_view(y);
```
Comment (mhoemmen) [inline tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_copy_kk.hpp:34]: You have the `Kokkos::View` and an execution space, so would you consider just using `Kokkos::deep_copy` here?

**Review by mhoemmen — CHANGES_REQUESTED** [review]

File: tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_copy_kk.hpp (hunk 1 above)
Comment (fnrizzi) [reply tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_copy_kk.hpp:34]: I think that if the layout of x and y are different, the deep copy fails. To use deep copy I would need to specialize for when the layouts are the same. Right?

Comment (fnrizzi) [reply tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas1_copy_kk.hpp:34]: done!

**Comment by mhoemmen** [comment]
Thanks so much!

**Pull request merged**